Sources/Features/Breadcrumb/BacktraceBreadcrumbFileHelper.swift

```diff
@@ -36,78 +32,75 @@ enum BacktraceBreadcrumbFileHelperError: Error {
+            // Check if breadcrumb size is larger than the maximum specified
+            if breadcrumbSize > maximumIndividualBreadcrumbSize {
+                BacktraceLogger.warning(
+                    "Discarding breadcrumb that was larger than the maximum specified (\(maximumIndividualBreadcrumbSize).")
+                return false
+            }
+            // Store breadcrumb Json String and size in Dictionary [String : Any]
+            let queueBreadcrumb = ["breadcrumbJson": breadcrumbJsonString, "size": breadcrumbSize] as [String : Any]
+            // Queue breacrumb
+            queue.enqueue(queueBreadcrumb)
+            // Iterate over the queue from newest to oldest breadcrumb and build an array of encoded strings
+            var breadcrumbsArray = [String]()
+            var size = 0
+            for index in (0..<queue.count).reversed() {
+                guard let queueBreadcrumb = queue.allElements()[index] as? [String: Any] else {
+                    BacktraceLogger.warning("Error weh fetching breacrumbs from queue")
+                    return false
+                }
+                guard let breadcrumbSize = queueBreadcrumb["size"] as? Int else {
+                    BacktraceLogger.warning("Error when adding breadcrumbSize to array")
+                    return false
+                }
+                // Pop last element if size id greater than maxQueueFileSizeBytes
+                if size + breadcrumbSize > maxQueueFileSizeBytes && !queue.isEmpty {
+                    queue.pop()
+                } else {
+                    guard let breadcrumbJsonData = queueBreadcrumb["breadcrumbJson"] as? String else {
+                        BacktraceLogger.warning("Error when adding breadcrumbJson to array")
+                        return false
+                    }
+                    breadcrumbsArray.append(breadcrumbJsonData)
+                    size += breadcrumbSize
                 }
-
-                try queue.add(textBytes, error: ())
             }
+            // Write breadcrumbs to file
+            let breadcrumbString = "[\(breadcrumbsArray.joined(separator: ","))]"
+            try breadcrumbString.write(to: self.breadcrumbLogURL, atomically: true, encoding: .utf8)
+            return true
         } catch {
-            BacktraceLogger.warning("\(error.localizedDescription) \nWhen adding breadcrumb to file")
+            BacktraceLogger.warning("Error when adding breadcrumb to file: \(error)")
             return false
         }
-
-        return true
     }
-
+    
     func clear() -> Bool {
-        do {
-            try dispatchQueue.sync {
-                try queue.clearAndReturnError()
-            }
-        } catch {
-            BacktraceLogger.warning("\(error.localizedDescription) \nWhen clearing breadcrumb file")
-            return false
+        dispatchQueue.sync {
+            queue.clear()
+            clearBreadcrumbLogFile(at:self.breadcrumbLogURL)
```

Comment:
It's ok to remove the try/catch here?

Reply:
yes since clearBreadcrumbLogFile() has it now.